### PR TITLE
[Enhancement] support using struct/map in mv

### PR DESCRIPTION
**Pull request by murphyatwork**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 25 / 28 (89.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|21|24|87.50%|[714, 822, 823]|
|:large_blue_circle:|com/starrocks/catalog/StructType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CTASAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|1|1|100.00%|[]|

**Review by Seaven — APPROVED** [review]

**Review by fzhedu — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzerUtils.java [hunk 1]
```diff
@@ -725,9 +698,20 @@ public static Type transformTypeForMv(Type srcType) {
...
+            StructType structType = (StructType) srcType;
+            List<StructField> mappingFields = structType.getFields().stream()
+                    .map(x -> new StructField(x.getName(), transformTableColumnType(x.getType(), convertDouble),
+                            x.getComment()))
+                    .collect(Collectors.toList());
+            newType = new StructType(mappingFields, structType.isNamed());
         } else {
-            throw new SemanticException("Unsupported Mv transform type: %s", srcType);
+            // Default behavior is keep source type
+            return newType;
```
Comment (LiShuMing) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzerUtils.java:714]: So MV/CTAS will support all types: like hll/bitmap/percentile? If so, may add tests for those?

 we can remove the restrictions for MV/CTAS in the doc later.

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 25 / 28 (89.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|21|24|87.50%|[714, 822, 823]|
|:large_blue_circle:|com/starrocks/catalog/StructType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CTASAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|1|1|100.00%|[]|

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 25 / 28 (89.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|21|24|87.50%|[714, 822, 823]|
|:large_blue_circle:|com/starrocks/catalog/StructType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CTASAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|1|1|100.00%|[]|

**Review by Seaven — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**